packages/pretty-format/src/__tests__/dom_collection.test.js

```diff
@@ -12,26 +12,111 @@
   DOMCollection,
+  DOMElement,
 ]);
 
 const expect: any = global.expect;
 expect.extend({toPrettyPrintTo});
 
-describe('DOMCollection Plugin', () => {
-  it('supports a DOMStringMap', () => {
+describe('DOMCollection plugin for object properties', () => {
+  it('supports DOMStringMap', () => {
     const el = document.createElement('div');
     el.dataset.foo = 'bar';
 
     expect(el.dataset).toPrettyPrintTo('DOMStringMap {\n  "foo": "bar",\n}');
   });
 
-  it('supports a NamedNodeMap', () => {
+  it('supports NamedNodeMap', () => {
     const el = document.createElement('div');
     el.setAttribute('foo', 'bar');
 
     expect(el.attributes).toPrettyPrintTo('NamedNodeMap {\n  "foo": "bar",\n}');
   });
 });
+
+describe('DOMCollection plugin for list items', () => {
+  const select = document.createElement('select');
+  select.innerHTML = [
+    '<option value="1">one</option>',
+    '<option value="2">two</option>',
+    '<option value="3">three</option>',
+  ].join('');
+
+  const form = document.createElement('form');
+  form.appendChild(select);
+
+  const expectedOption1 = [
+    '  <option',
+    '    value="1"',
+    '  >',
+    '    one',
+    '  </option>,', // comma because item
+  ].join('\n');
+  const expectedOption2 = [
+    '  <option',
+    '    value="2"',
+    '  >',
+    '    two',
+    '  </option>,', // comma because item
+  ].join('\n');
+  const expectedOption3 = [
+    '  <option',
+    '    value="3"',
+    '  >',
+    '    three',
+    '  </option>,', // comma because item
+  ].join('\n');
+
+  const expectedHTMLCollection = [
+    'HTMLCollection [',
+    expectedOption1,
+    expectedOption2,
+    expectedOption3,
+    ']',
+  ].join('\n');
+
+  it('supports HTMLCollection for getElementsByTagName', () => {
+    const options = form.getElementsByTagName('option');
+    expect(options).toPrettyPrintTo(expectedHTMLCollection);
+  });
+
+  it('supports HTMLCollection for children', () => {
+    expect(select.children).toPrettyPrintTo(expectedHTMLCollection);
+  });
+
+  const expectedNodeList = [
+    'NodeList [',
+    expectedOption1,
+    expectedOption2,
+    expectedOption3,
+    ']',
+  ].join('\n');
+
+  it('supports NodeList for querySelectorAll', () => {
+    const options = form.querySelectorAll('option');
+    expect(options).toPrettyPrintTo(expectedNodeList);
+  });
+
+  it('supports NodeList for childNodes', () => {
+    expect(select.childNodes).toPrettyPrintTo(expectedNodeList);
+  });
+
+  const expectedHTMLOptionsCollection = [
+    'HTMLOptionsCollection [',
+    expectedOption1,
+    expectedOption2,
+    expectedOption3,
+    ']',
+  ].join('\n');
+
+  it('supports HTMLOptionsCollection for options', () => {
+    expect(select.options).toPrettyPrintTo(expectedHTMLOptionsCollection);
+  });
+
+  // Omitted a test for form.elements because constructor.name
+  // is HTMLCollection in jsdom version 11
+  // might become HTMLFormControlsCollectio in jsdom 12
```

Comment:
jsdom 12 is released, and while we won't upgrade for some time, maybe we can normalize the name to `HTMLFormControlsCollection`?

Reply:
Verified outside Jest that `form.elements` still `HTMLCollection` class in jsdom@12.2.0 so added the test. Wrote a comment so that we will know to update the criterion if it ever changes.